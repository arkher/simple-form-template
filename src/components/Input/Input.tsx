import React, { LegacyRef, forwardRef } from "react";

const Input = forwardRef(
  (
    {
      value,
      onChange,
      onBlur,
      ...props
    }: React.InputHTMLAttributes<HTMLInputElement>,
    ref?: LegacyRef<HTMLInputElement>
  ) => {
    return (
      <input
        ref={ref}
        className="bg-white mb-8 border-2 border-gray-500 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-4"
        value={value}
        onChange={onChange}
        onBlur={onBlur}
        data-testid="input"
        {...props}
      />
    );
  }
);

export default Input;
