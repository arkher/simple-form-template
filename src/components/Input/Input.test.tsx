import { fireEvent, render, screen } from "@testing-library/react";
import Input from "./Input";
import "@testing-library/jest-dom/extend-expect";

describe("Input", () => {
  it("should render", () => {
    render(<Input data-testid="input" />);
    expect(screen.getByTestId("input")).toBeTruthy();
  });

  it("should change text correctly when typing", () => {
    render(<Input data-testid="input" />);
    fireEvent.change(screen.getByTestId("input"), {
      target: { value: "test" },
    });
    expect(screen.getByTestId("input")).toHaveValue("test");
  });

  it("should match snapshot", () => {
    render(<Input data-testid="input" />);
    expect(screen.getByTestId("input")).toMatchSnapshot();
  });
});
