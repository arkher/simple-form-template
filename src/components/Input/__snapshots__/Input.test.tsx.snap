// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<input
  class="bg-white mb-8 border-2 border-gray-500 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-4"
  data-testid="input"
  value=""
/>
`;
